From stable-bounces@linux.kernel.org Fri Jul 14 15:58:35 2006
Date: Fri, 14 Jul 2006 18:51:41 -0400
From: Chuck Ebbert <76306.1226@compuserve.com>
To: linux-netdev <netdev@vger.kernel.org>
Message-ID: <200607141855_MC3-1-C4FF-BD1B@compuserve.com>
Content-Disposition: inline
Cc: Andrew Morton <akpm@osdl.org>, Toralf Foerster <toralf.foerster@gmx.de>,
        linux-stable <stable@kernel.org>,
        linux-kernel <linux-kernel@vger.kernel.org>,
        "John W. Linville" <linville@tuxdriver.com>
Subject: ieee80211: TKIP requires CRC32

ieee80211_crypt_tkip will not work without CRC32.

  LD      .tmp_vmlinux1
net/built-in.o: In function `ieee80211_tkip_encrypt':
net/ieee80211/ieee80211_crypt_tkip.c:349: undefined reference to `crc32_le'

Reported by Toralf Foerster <toralf.foerster@gmx.de>

Signed-off-by: Chuck Ebbert <76306.1226@compuserve.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 net/ieee80211/Kconfig |    1 +
 1 file changed, 1 insertion(+)

--- linux-2.6.17.6.orig/net/ieee80211/Kconfig
+++ linux-2.6.17.6/net/ieee80211/Kconfig
@@ -58,6 +58,7 @@ config IEEE80211_CRYPT_TKIP
 	depends on IEEE80211 && NET_RADIO
 	select CRYPTO
 	select CRYPTO_MICHAEL_MIC
+	select CRC32
 	---help---
 	Include software based cipher suites in support of IEEE 802.11i
 	(aka TGi, WPA, WPA2, WPA-PSK, etc.) for use with TKIP enabled
